opm/simulators/wells/VFPHelpers.hpp

```diff
@@ -560,284 +167,16 @@ bool hasTable(const std::map<int, std::reference_wrapper<const T>> tables, int t
-        int i=0;
-        bool found = false;
-        for (; i<nthp-1; ++i) {
-            const double& y0 = bhp_array[i  ];
-            const double& y1 = bhp_array[i+1];
-
-            if (y0 < bhp && bhp <= y1) {
-                found = true;
-                break;
-            }
-        }
-        if (found) {
-            const double& x0 = thp_array[i  ];
-            const double& x1 = thp_array[i+1];
-            const double& y0 = bhp_array[i  ];
-            const double& y1 = bhp_array[i+1];
-            thp = detail::findX(x0, x1, y0, y1, bhp);
-        }
-        else if (bhp <= bhp_array[0]) {
-            //TODO: LOG extrapolation
-            const double& x0 = thp_array[0];
-            const double& x1 = thp_array[1];
-            const double& y0 = bhp_array[0];
-            const double& y1 = bhp_array[1];
-            thp = detail::findX(x0, x1, y0, y1, bhp);
-        }
-        //Target bhp greater than all values in array, extrapolate
-        else if (bhp > bhp_array[nthp-1]) {
-            //TODO: LOG extrapolation
-            const double& x0 = thp_array[nthp-2];
-            const double& x1 = thp_array[nthp-1];
-            const double& y0 = bhp_array[nthp-2];
-            const double& y1 = bhp_array[nthp-1];
-            thp = detail::findX(x0, x1, y0, y1, bhp);
-        }
-        else {
-            OPM_THROW(std::logic_error, "Programmer error: Unable to find THP in THP array");
-        }
-    }
-
-    return thp;
-}
-
-
-
-// a data type use to do the intersection calculation to get the intial bhp under THP control
-struct RateBhpPair {
-    double rate;
-    double bhp;
-};
-
-
-// looking for a intersection point a line segment and a line, they are both defined with two points
-// it is copied from #include <opm/polymer/Point2D.hpp>, which should be removed since it is only required by the lagacy polymer
-inline bool findIntersection(const std::array<RateBhpPair, 2>& line_segment, const std::array<RateBhpPair, 2>& line, double& bhp) {
-    const double x1 = line_segment[0].rate;
-    const double y1 = line_segment[0].bhp;
-    const double x2 = line_segment[1].rate;
-    const double y2 = line_segment[1].bhp;
-
-    const double x3 = line[0].rate;
-    const double y3 = line[0].bhp;
-    const double x4 = line[1].rate;
-    const double y4 = line[1].bhp;
-
-    const double d = (x1 - x2) * (y3 - y4) - (y1 - y2) * (x3 - x4);
-
-    if (d == 0.) {
-        return false;
-    }
-
-    const double x = ((x3 - x4) * (x1 * y2 - y1 * x2) - (x1 - x2) * (x3 * y4 - y3 * x4)) / d;
-    const double y = ((y3 - y4) * (x1 * y2 - y1 * x2) - (y1 - y2) * (x3 * y4 - y3 * x4)) / d;
-
-    if (x >= std::min(x1,x2) && x <= std::max(x1,x2)) {
-        bhp = y;
-        return true;
-    } else {
-        return false;
-    }
-}
-
-// calculating the BHP from thp through the intersection of VFP curves and inflow performance relationship
-inline bool findIntersectionForBhp(const std::vector<RateBhpPair>& ratebhp_samples,
```

Comment:
Just a note, this is the old implementation regarding to how to calculate the BHP from the intersection of the IPR and VFP curves, which is the standard way in textbook and manual. OPM is going to a different direct now, and looks like working.  In the future, we might come back to revisit this method with more information. It is not used now and okay to be removed for now.